src/test/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformationGetterTest.java

```diff
@@ -1,29 +1,71 @@
 package co.rsk.federate.signing.hsm.message;
 
+import static co.rsk.federate.bitcoin.BitcoinTestUtils.coinListOf;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.Mockito.anyLong;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
 import co.rsk.bitcoinj.core.Sha256Hash;
 import co.rsk.crypto.Keccak256;
+import co.rsk.federate.bitcoin.BitcoinTestUtils;
 import co.rsk.federate.signing.utils.TestUtils;
 import co.rsk.peg.BridgeEvents;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import org.bouncycastle.util.encoders.Hex;
 import org.ethereum.core.*;
 import org.ethereum.db.*;
 import org.ethereum.vm.DataWord;
 import org.ethereum.vm.LogInfo;
 import org.ethereum.vm.PrecompiledContracts;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 class ReleaseCreationInformationGetterTest {
 
+    private BtcTransaction pegoutBtcTx;
+    private Transaction pegoutCreationRskTx;
+    private Block pegoutCreationBlock;
+    private TransactionReceipt pegoutCreationRskTxReceipt;
+    private TransactionInfo pegoutCreationRskTxInfo;
+
+    @BeforeEach
+    void setUp() {
+        Sha256Hash pegoutBtcTxHash = BitcoinTestUtils.createHash(1);
+        pegoutBtcTx = mock(BtcTransaction.class);
+        when(pegoutBtcTx.getHash()).thenReturn(pegoutBtcTxHash);
+
+        Keccak256 pegoutCreationRskTxHash = TestUtils.createHash(2);
+        pegoutCreationRskTx = mock(Transaction.class);
+        when(pegoutCreationRskTx.getHash()).thenReturn(pegoutCreationRskTxHash);
+        when(pegoutCreationRskTx.getReceiveAddress()).thenReturn(PrecompiledContracts.BRIDGE_ADDR);
+
+        Keccak256 pegoutCreationBlockHash = TestUtils.createHash(3);
```

Comment:
You can achieve this using BlockHeaderBuilder to create actual Block objects instead of mocks. Check https://github.com/rsksmart/powpeg-node/blob/master/src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImplTest.java#L636 

Just an idea

Reply:
Done